layouts/wallet.vue

```diff
@@ -17,51 +25,87 @@ import logTrackerEvent, { setLoggerUser } from '~/utils/logger'
 
 
 const walletModule = namespace('wallet')
+const bookApiModule = namespace('book-api')
+
 
 @Component
 export default class WalletLayout extends Vue {
   @walletModule.Getter('getWalletAddress') walletAddress!: string
+  @walletModule.Action('disconnectWallet') disconnectWallet!: () => void
   @walletModule.Action('openConnectWalletModal') openConnectWalletModal!: (params: { language: string, fullPath?: string }) => Promise<any>
   @walletModule.Action('initWallet') initWallet!: (params: { method: any, accounts: any, offlineSigner?: any }) => Promise<any>
-  @walletModule.Action('restoreSessionIfNecessary') restoreSessionIfNecessary!: () => Promise<any>
-
+  @walletModule.Action('signMessageMemo') signMessageMemo!: (action: string, permissions?: string[]) => Promise<any>
+  @walletModule.Getter('getSigner') signer!: any
+  @bookApiModule.Getter('getSessionWallet') sessionWallet!: string
+  @bookApiModule.Action('restoreSession') restoreSession!: () => void
+  @bookApiModule.Action('authenticate') authenticate!: ({ inputWallet, signature }: { inputWallet?: string, signature?: any }) => Promise<any>
+  @bookApiModule.Action('clearSession') clearSession!: () => void
 
   @walletModule.Action toggleAlert!: (
     isShow: boolean,
     error: string,
     severity: string
   ) => void
 
+  isSignInLoading = false
+
   get isHideFooter() {
     return this.$route.path.includes('/nft/purchase/') || this.$route.query.layout === 'popup';
   }
 
   async mounted() {
-    await this.restoreSessionIfNecessary()
-    if (!this.walletAddress) {
+    await this.restoreSession()
+    if (!this.sessionWallet) {
+      this.isSignInLoading = true
       const connection = await this.openConnectWalletModal({
         language: this.$i18n.locale.split('-')[0],
         fullPath: this.$route.fullPath,
       })
       // Re-login
-      if (connection) {
-        const { method, accounts } = connection
-        logTrackerEvent(
-          this,
-          'user',
-          `connected_wallet_${method}`,
-          'connected_wallet',
-          1,
-        )
-        setLoggerUser(this, {
-          wallet: accounts[0].address,
-          method,
-        })
-        return this.initWallet(connection)
+      try {
+        if (connection) {
+          const { method, accounts } = connection
+          logTrackerEvent(
+            this,
+            'user',
+            `connected_wallet_${method}`,
+            'connected_wallet',
+            1,
+          )
+          setLoggerUser(this, {
+            wallet: accounts[0].address,
+            method,
+          })
+          await this.initWallet(connection)
+          if (!this.walletAddress || !this.signer) return
+          const signature = await this.signMessageMemo('authorize', [
+            'profile',
+            'read:nftbook',
+            'write:nftbook',
+            'read:nftcollection',
+            'write:nftcollection',
+          ])
+          if (!signature) {
+            return
```

Comment:
Prefer throwing error for ui display

Reply:
[🥅 Throw connection error](https://github.com/likecoin/app-like-co/pull/474/commits/cc9826077540dfdcd12ebe0f57470e7419f8c922)